src/Blueprint.h

```diff
@@ -357,10 +340,18 @@ namespace snowcrash {
             UndefinedElement = 0, // Unknown
             CategoryElement,      // Group of other elements
             CopyElement,          // Human readable text
+            AssetElement,         // Asset of API description
             ResourceElement,      // Resource
             DataStructureElement  // Data Structure
         };
 
+        /** Role of an element */
+        enum Role {
```

Comment:
This is indeed a very interesting think to ponder about. I was considering creating the `Asset` sub-classes `Schema` and `BodyExample` but I believe there can be an infinite number of assets roles and media types in the API while there will always be only two roles for a message-payload. 

To be frank I do not have any solid ground for deciding role vs defining a new element. What I know tho is that it is safer to define the attributes on the specific-element level then general-element level. 

In the other words: think about the rules of the extending – it will be easier to add something optional to the element base attributes sets later, than remove it from the set.